archivy/check_changes.py

```diff
@@ -63,7 +63,7 @@ def on_modified(self, event):
             filename = event.src_path.split(SEP)[-1]
             if re.match(DATAOBJ_REGEX, filename) and self.ELASTIC:
                 self.app.logger.info(f"Detected changes to {event.src_path}")
-                dataobj = models.DataObj.from_file(event.src_path)
+                dataobj = models.DataObj.from_md(open(event.src_path).read())
```

Comment:
Potential memory leak, and unhandled file close. This can cause data lose

Reply:
Sorry, Python isn't my first language, how should I fix this?

Reply:
Hmm,

files can be opened in two ways

```bash
with open("foobar") as fp:
    data = fp.read()
```
or 
```bash
a = open("foobar")
data = a.read()
a.close()
```

in your current code, you open the file, but never close it; the file remains on the heap, until closed; this is not a python specific problem; its basically GNU/Linux/BSD;

Reply:
Ok, I just meant that I wasn't sure what the right way to close the file was in Python. Thanks

Reply:
Learning C can help understand what happens under the hood, in the python syscalls. open files remain on the heap; and more and more open files over time cause high memory usage. In `C` we do something like `free()` after calling `malloc()`;

Reply:
The memory leakage issue should be fixed now right?

Reply:
yes, better :smile:

Reply:
Good, do you see any other issues?

I've mostly used C++ and not C (mostly for competitive programming), so I have a bit less experience with scripting on the lower-level :joy:

Reply:
Me too; I started learning C and C++ recently; and yes; I realized very late that I was not learning anything new from competitive programming rather than trying to improve my math; so I quit CP; and moved to functional programming :smiley: 

> Good, do you see any other issues?

I do not see any other issues; but, two brains are always better than one; we need more reviewers though :(

Reply:
True...

We'll prob get more vision once we launch 1.0

> moved to functional programming

I'm learning it too! With haskell, what about you?